roles/matrix-bridge-heisenbridge/tasks/setup_install.yml

```diff
@@ -0,0 +1,56 @@
+---
+
+- name: Ensure heisenbridge image is pulled
+  docker_image:
+    name: "{{ matrix_heisenbridge_docker_image }}"
+    source: "{{ 'pull' if ansible_version.major > 2 or ansible_version.minor > 7 else omit }}"
+
+- name: Ensure heisenbridge paths exist
+  file:
+    path: "{{ item }}"
+    state: directory
+    mode: 0750
+    owner: "{{ matrix_user_username }}"
+    group: "{{ matrix_user_groupname }}"
+  with_items:
+    - "{{ matrix_heisenbridge_registration_path }}"
+
+- name: Check if heisenbridge registration.yaml exists
+  stat:
+    path: "{{ matrix_heisenbridge_registration_path }}/registration.yaml"
+  register: matrix_heisenbridge_registration_yaml_stat
+
+- name: Ensure heisenbridge registration.yaml installed if provided
+  copy:
+    content: "{{ matrix_heisenbridge_registration|to_nice_yaml }}"
+    dest: "{{ matrix_heisenbridge_registration_path }}/registration.yaml"
+    mode: 0644
+    owner: "{{ matrix_user_username }}"
+    group: "{{ matrix_user_groupname }}"
+  when: not matrix_heisenbridge_registration_yaml_stat.stat.exists and matrix_heisenbridge_registration|length > 0
+
+- name: Generate heisenbridge registration.yaml if no yaml provided
+  shell: >-
+    {{ matrix_host_command_docker }} run --rm --name heisenbridge-registration
+    --user={{ matrix_user_uid }}:{{ matrix_user_gid }}
+    --cap-drop=ALL
+    --mount type=bind,src={{ matrix_heisenbridge_registration_path }},dst=/config
+    -w /config
+    {{ matrix_heisenbridge_docker_image }}
+    --config /config/registration.yaml
+    --listen-address heisenbridge
+    --listen-port 9898
+    --generate
+  when: not matrix_heisenbridge_registration_yaml_stat.stat.exists and matrix_heisenbridge_registration|length == 0
```

Comment:
I'm okay with having a template registration file and not generating it like it currently does as long as it doesn't require user intervention (like setting up tokens manually).

`@irc_.*` is correct but the prefix is free to be changed and it is respected by the bridge. It doesn't currently support the domain part when reading it in but that's fixable.

The only issue I have with `@heisenbridge_` is that it makes the MXIDs _very_ long but if it's made configurable and overridable back to `@irc_` if wanted I guess that's fine to avoid conflicts in the default configuration.

Reply:
The tokens are generated automatically derived from other secrets (see my review comment in `group_vars/matrix_servers`), so no manual intervention is required. The registration file can be fully rebuilt every time without a problem.

How about `@irch_` for a non-conflicting (and still short) prefix?

Alternatively, we can continue using `@irc_` and put up a warning that it will conflict with `matrix-appservice-irc`. It's probably better if one can enable any/all bridges without a conflict by default though.

Reply:
I have addressed this now with `@hb_`. If you think it has a high chance of conflicting with local users we can set it to `@irch_`.

Reply:
It's probably okay. Although.. you're right, `@hb_` may have a higher chance of being used by real users compared to something like `@irch_` or such a prefix that we use for other bridges (real users are less likely to name themselves `@discord_....`).